api/post.go

```diff
@@ -916,18 +916,53 @@ func getPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	pchan := Srv.Store.Post().GetPosts(id, offset, limit)
-
-	if result := <-pchan; result.Err != nil {
+	var postList *model.PostList
+	if result := <-Srv.Store.Post().GetPosts(id, offset, limit); result.Err != nil {
 		c.Err = result.Err
 		return
 	} else {
-		list := result.Data.(*model.PostList)
+		postList = result.Data.(*model.PostList)
+	}
 
-		w.Header().Set(model.HEADER_ETAG_SERVER, etag)
-		w.Write([]byte(list.ToJson()))
+	setProfilesForPostList(c, postList)
+	if c.Err != nil {
+		return
 	}
 
+	w.Header().Set(model.HEADER_ETAG_SERVER, etag)
+	w.Write([]byte(postList.ToJson()))
+
+}
+
+func setProfilesForPostList(c *Context, postList *model.PostList) {
+	userIds := []string{}
+	used := map[string]bool{}
+	for _, p := range postList.Posts {
+		if _, found := used[p.UserId]; !found {
+			userIds = append(userIds, p.UserId)
+			used[p.UserId] = true
+		}
+	}
+
+	if len(userIds) > 0 {
+		if result := <-Srv.Store.User().GetProfilesFromList(userIds); result.Err != nil {
+			c.Err = result.Err
+			return
+		} else {
+			profiles := result.Data.(model.UserMap)
+
+			options := utils.Cfg.GetSanitizeOptions()
+			options["passwordupdate"] = false
+			if c.IsSystemAdmin() {
+				options["fullname"] = true
+				options["email"] = true
+			}
+
+			(&profiles).ClearNonProfileFields(options)
```

Comment:
Couldn't this just be `profiles.ClearNonProfileFields(options)`? I thought Go handles whether or not method receivers are pointers automatically